cosmetics-web/app/validators/accept_and_submit_validator.rb

```diff
@@ -1,9 +1,43 @@
 class AcceptAndSubmitValidator < ActiveModel::Validator
-  def validate(record)
-    missing_nano_materials = record.missing_nano_materials
+  def validate(notification)
+    validate_nano_materials(notification)
+    validate_image_uploads(notification)
+    validate_frame_formulation_uploads(notification)
+  end
+
+  def validate_nano_materials(notification)
+    missing_nano_materials = notification.missing_nano_materials
     if missing_nano_materials.present?
       missing_nano_materials.each do |nano|
-        record.errors.add :base, "#{nano.name} is not included in any items"
+        notification.errors.add :nano_materials, "#{nano.name} is not included in any items"
+      end
+    end
+  end
+
+  def validate_image_uploads(notification)
+    notification.image_uploads.each do |image_upload|
+      if image_upload.pending_antivirus_check?
+        notification.errors.add :image_uploads, "Image #{image_upload.file.filename} is still being processed"
+      end
+      if image_upload.failed_antivirus_check?
+        notification.errors.add :image_uploads, "Image #{image_upload.file.filename} failed antivirus check. Remove image and try again"
+      end
+    end
+    if notification.image_uploads.blank?
+      notification.errors.add :image_uploads, "Product image is missing"
+    end
+  end
+
+  def validate_frame_formulation_uploads(notification)
+    notification.components.each do |component|
+      if component.formulation_file_pending_antivirus_check?
+        notification.errors.add :formulation_uploads, "File #{component.formulation_file.filename} is still being processed"
+      end
+      if component.formulation_file_failed_antivirus_check?
+        notification.errors.add :formulation_uploads, "File #{component.formulation_file.filename} failed antivirus check. Remove file and try again"
+      end
+      if component.formulation_required?
+        notification.errors.add :formulation_uploads, "Item #{component.name} is missing formulation file"
       end
```

Comment:
:raising_hand: mutually exclusive conditions:
```suggestion
      if component.formulation_required?
        notification.errors.add :formulation_uploads, "Item #{component.name} is missing formulation file"
      elsif component.formulation_file_pending_antivirus_check?
        notification.errors.add :formulation_uploads, "File #{component.formulation_file.filename} is still being processed"
      elsif component.formulation_file_failed_antivirus_check?
        notification.errors.add :formulation_uploads, "File #{component.formulation_file.filename} failed antivirus check. Remove file and try again"
      end
```